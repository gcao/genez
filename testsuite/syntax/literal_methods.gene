# Test method calls on literals

# String literal methods
(println ("hello" .length))

# Number literal methods
(println (10 .+ 5))
(println (10 .- 3))
(println (10 .* 2))
(println (10 ./ 5))

# Float literal methods
(println (3.5 .+ 1.5))
(println (4.0 .* 2.5))

# Array literal methods
(println ([1 2 3 4] .length))
(println ([10 20 30] .at 1))
(println ([] .length))

