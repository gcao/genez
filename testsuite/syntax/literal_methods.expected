5
15
7
20
2
5
10
4
20
0